cmake/modules/AddSwift.cmake

```diff
@@ -916,10 +916,17 @@ function(_add_swift_library_single target name)
   set(prefixed_link_libraries)
   foreach(dep ${SWIFTLIB_SINGLE_LINK_LIBRARIES})
     if("${dep}" MATCHES "^clang")
-      set(dep "${LLVM_LIBRARY_OUTPUT_INTDIR}/lib${dep}.a")
-    endif()
-    if("${dep}" STREQUAL "cmark")
-      set(dep "${CMARK_LIBRARY_DIR}/lib${dep}.a")
+      if ("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "WINDOWS")
+        set(dep "${LLVM_LIBRARY_OUTPUT_INTDIR}/${dep}.lib")
+      else()
+        set(dep "${LLVM_LIBRARY_OUTPUT_INTDIR}/lib${dep}.a")
+      endif()
+    elseif("${dep}" STREQUAL "cmark")
+      if ("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "WINDOWS")
+        set(dep "${CMARK_LIBRARY_DIR}/cmark.lib")
+      else()
+        set(dep "${CMARK_LIBRARY_DIR}/lib${dep}.a")
+      endif()
```

Comment:
I'm not sure we adjust the prefix accordingly: the layout of the cmark build directory on Windows is as follows:
- cmark.dll
- cmark.exe
- cmark.exp
- cmark.ilk
- cmark.lib
- cmark.pdb
- cmark_dll.pdb
- cmark_export.h
- cmark_static.lib
- libcmark.pc

There is no such file "libcmark.a", so we do in fact need to adjust the depency for Windows.